tests/AvState_test.cpp

```diff
@@ -14,29 +12,66 @@ int main(int argc, char** argv) {
     DataDump dump;
 
     // Initial state should be INIT
-    fsm.update(dump);
     assert(fsm.getCurrentState() == State::INIT);
```

Comment:
Yes, check first after constructor call, but might also be good to check after the first `fsm.update()` the state remains `INIT`.